### PR TITLE
nicer declarative .async boundaries and remove Module.nest()

### LINKED ISSUE #19732: A more consistent and succinct way to mark async boundaries
`.withAttributes(Attributes.asyncBoundary)` is verbose given that it is something one would expect to do pretty often. I propose that we introduce a shortcut `{Source,Flow,Sink}.async` just like `.named()` which exists already and remove `Flow.viaAsync(flow)` which will then instead be `Flow.via(flow.async)`

### PR DESCRIPTION
**Pull request by rkuhn**
fixes #19732 

This adds the nest() removal on top of Johan’s work.

### CODE REVIEW

File: akka-stream-tests/src/test/resources/reference.conf [hunk 1]
```diff
@@ -8,4 +8,5 @@ akka {
   akka.actor.warn-about-java-serializer-usage = false
 
   stream.materializer.debug.fuzzing-mode = on
+  stream.secret-test-fuzzing-warning-disable = 42
```
Comment (drewhk) [inline akka-stream-tests/src/test/resources/reference.conf]: ?

File: akka-stream-tests/src/test/scala/akka/stream/impl/fusing/InterpreterSpec.scala [hunk 2]
```diff
@@ -511,7 +511,9 @@ class InterpreterSpec extends AkkaSpec with GraphInterpreterSpecKit {
       downstream.requestOne()
       lastEvents() should be(Set(RequestOne))
 
-      upstream.onComplete()
+      EventFilter[IllegalArgumentException](pattern = ".*Cannot pull closed port.*", occurrences = 1).intercept {
```
Comment (drewhk) [inline akka-stream-tests/src/test/scala/akka/stream/impl/fusing/InterpreterSpec.scala]: why the change?

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowIteratorSpec.scala [hunk 3]
```diff
@@ -40,7 +40,9 @@ class FlowIterableSpec extends AbstractFlowIteratorSpec {
     sub.request(1)
     c.expectNext(1)
     c.expectNoMsg(100.millis)
-    sub.request(2)
+    EventFilter[IllegalStateException](message = "not two", occurrences = 1).intercept {
```
Comment (drewhk) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowIteratorSpec.scala]: same here

File: akka-stream-tests/src/test/resources/reference.conf (hunk 1 above)
Comment (rkuhn) [inline akka-stream-tests/src/test/resources/reference.conf]: cleaning up the build output

**Comment by drewhk** [comment]
LGTM after those questions are explalined.

File: akka-stream-tests/src/test/scala/akka/stream/impl/fusing/InterpreterSpec.scala (hunk 2 above)
Comment (rkuhn) [inline akka-stream-tests/src/test/scala/akka/stream/impl/fusing/InterpreterSpec.scala]: same as above (and below)

File: akka-stream-tests/src/test/resources/reference.conf (hunk 1 above)
Comment (2beaucoup) [inline akka-stream-tests/src/test/resources/reference.conf]: A `FuzzingGraphInterpreter` that lives in stream-tests would be nice.

Comment (drewhk) [inline akka-stream-tests/src/test/resources/reference.conf]: WDYM? It is already exposed to users: http://doc.akka.io/docs/akka-stream-and-http-experimental/2.0.3/scala/stream-testkit.html#fuzzing-mode

Comment (2beaucoup) [inline akka-stream-tests/src/test/resources/reference.conf]: I see. I thought that way it would be more obvious that fuzzing is only intended for testing and we could get rid of the warning/setting.

(Correction: I guess stream-testkit ISO stream-tests would be the right place.)

Comment (drewhk) [inline akka-stream-tests/src/test/resources/reference.conf]: There is a big warning in the docs:

"Warning
Never use this setting in production or benchmarks. This is a testing tool to provide more coverage of your code during tests, but it reduces the throughput of streams. A warning message will be logged if you have this setting enabled."

Also in the reference.conf:

"       # Enables the fuzzing mode which increases the chance of race conditions
        # by aggressively reordering events and making certain operations more
        # concurrent than usual.
        # This setting is for testing purposes, NEVER enable this in a production
        # environment!
        # To get the best results, try combining this setting with a throughput
        # of 1 on the corresponding dispatchers."

Since these are the two places where ordinary people can learn about the existence of the feature, I assume that these warnings, and additionally the warning log, is more than enough.

Comment (2beaucoup) [inline akka-stream-tests/src/test/resources/reference.conf]: OK. I just thought that this approach would be cleaner as it removes testing knobs from the API/settings. But the current warnings should work as well.

> Since these are the two places where ordinary people can learn about the existence of the feature...

A third place would be the API itself. ;)
